src/stdlib_str2num.f90

```diff
@@ -0,0 +1,532 @@
+        ! -- In/out Variables
+        character(len=:), pointer :: s !> input string
+        real(sp), intent(in) :: mold !> dummy argument to disambiguate at compile time the generic interface
+        real(sp) :: r    !> Output real value
+        integer(1),intent(inout), optional :: stat
+        ! -- Internal Variables
+        integer(1) :: p !> position within the number
+        integer(1) :: err
+        !----------------------------------------------
+        call to_num_base(s,r,p,err)
+        p = min( p , len(s) )
+        s => s(p:)
+        if(present(stat)) stat = err
+    end function
+
+    elemental function to_double(s,mold) result(r)
+        ! -- In/out Variables
+        character(*), intent(in) :: s !> input string
+        real(dp), intent(in) :: mold !> dummy argument to disambiguate at compile time the generic interface
+        real(dp) :: r    !> Output real value
+        ! -- Internal Variables
+        integer(1) :: p !> position within the number
+        integer(1) :: stat ! error status
+        !----------------------------------------------
+        call to_num_base(s,r,p,stat)
+    end function
+    
+    function to_double_p(s,mold,stat) result(r)
+        ! -- In/out Variables
+        character(len=:), pointer :: s !> input string
+        real(dp), intent(in) :: mold !> dummy argument to disambiguate at compile time the generic interface
+        real(dp) :: r    !> Output real value
+        integer(1),intent(inout), optional :: stat
+        ! -- Internal Variables
+        integer(1) :: p !> position within the number
+        integer(1) :: err
+        !----------------------------------------------
+        call to_num_base(s,r,p,err)
+        p = min( p , len(s) )
+        s => s(p:)
+        if(present(stat)) stat = err
+    end function
+
+    function to_quad(s,mold) result(r)
+        ! -- In/out Variables
+        character(*), intent(in) :: s !> input string
+        real(qp), intent(in) :: mold !> dummy argument to disambiguate at compile time the generic interface
+        real(qp) :: r    !> Output real value
+        ! -- Internal Variables
+        integer(1) :: p !> position within the number
+        integer(1) :: stat ! error status
+        !----------------------------------------------
+        call to_num_base(s,r,p,stat)
+    end function
+    
+    function to_quad_p(s,mold,stat) result(r)
+        ! -- In/out Variables
+        character(len=:), pointer :: s !> input string
+        real(qp), intent(in) :: mold !> dummy argument to disambiguate at compile time the generic interface
+        real(qp) :: r    !> Output real value
+        integer(1),intent(inout), optional :: stat
+        ! -- Internal Variables
+        integer(1) :: p !> position within the number
+        integer(1) :: err
+        !----------------------------------------------
+        call to_num_base(s,r,p,err)
+        p = min( p , len(s) )
+        s => s(p:)
+        if(present(stat)) stat = err
```

Comment:
Note: Not all compilers support `qp`. Therefore, this should be preprocessed with `fypp`.
Using `stdlib_kinds` insead of `iso_fortran_env` will help to get the correctkind.

Reply:
Which preprocessing macro would you suggest here? I know that nvfortran does not support qp, but gfortran and ifort/ifx do.

Reply:
Solved it with "#:if WITH_QP"